c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -614,6 +753,13 @@ class DeviceCachingAllocator {
     } else if (block->pool == &large_blocks) {
       return remaining > kSmallSize;
     } else {
+      for (auto& gp : graph_pools) {
```

Comment:
This definitely looks very perf fishy

Reply:
It is perf fishy, but for common eager mode allocations we always return before reaching the loop. The loop should only be reached when allocating from a private pool, in other words, while capturing. The only perf danger i see here is I've increased the code size in a fairly hot function which might increase icache pressure. No idea if that effect is tangible.

Reply:
new `BlockPool.is_small` field simplifies this function.